src/components/map.js

```diff
@@ -281,108 +361,239 @@ let Map = class Map extends React.Component {
-        return;
-      }
-      const comparisonFeatures = {type: "FeatureCollection", features}
-      const [minX, minY, maxX, maxY] = turf.bbox(comparisonFeatures)
-      this.map.fitBounds(
-        [[minX, minY], [maxX, maxY]],
-        {
-          padding: {top: 20, bottom: 40, left: 20, right: 20},
-          animate: false,
-        }
-      )
-      // If sa2-comp doesn't exist then the map hasn't finished loading yet.
-      // In that case, ignore this call to highlightComparisonFeatures and let
-      // the map's on("load") call it after it has loaded all styles and
-      // sources.
-      const source = this.map.getSource("sa2-comp");
-      if (source) {
-        source.setData(comparisonFeatures)
+    if (features.length <= 0) {
+      return;
+    }
+    const comparisonFeatures = { type: "FeatureCollection", features };
+    const [minX, minY, maxX, maxY] = turf.bbox(comparisonFeatures);
+    this.map.fitBounds(
+      [
+        [minX, minY],
+        [maxX, maxY],
+      ],
+      {
+        padding: { top: 20, bottom: 40, left: 20, right: 20 },
+        animate: false,
       }
-  }
+    );
+    // If sa2-comp doesn't exist then the map hasn't finished loading yet.
+    // In that case, ignore this call to highlightComparisonFeatures and let
+    // the map's on("load") call it after it has loaded all styles and
+    // sources.
+    const source = this.map.getSource("sa2-comp");
+    if (source) {
+      source.setData(comparisonFeatures);
+    }
+  };
+
+  drawBusinessClusters() {
+    if (this.map.getLayer("sa2-fills")) this.map.removeLayer("sa2-fills");
+    if (!this.map.getLayer("clusters"))
+      this.map.addLayer({
+        id: "clusters",
+        type: "circle",
+        source: "business",
+        filter: ["has", "point_count"],
+        paint: {
+          // Use step expressions (https://docs.mapbox.com/mapbox-gl-js/style-spec/#expressions-step)
+          // with three steps to implement three types of circles:
+          //   * Blue, 20px circles when point count is less than 100
+          //   * Yellow, 30px circles when point count is between 100 and 750
+          //   * Pink, 40px circles when point count is greater than or equal to 750
+          "circle-color": [
+            "step",
+            ["get", "point_count"],
+            "#51bbd6",
+            100,
+            "#f1f075",
+            750,
+            "#f28cb1",
+          ],
+          "circle-radius": [
+            "step",
+            ["get", "point_count"],
+            20,
+            100,
+            30,
+            750,
+            40,
+          ],
+        },
+      });
+
+    if (!this.map.getLayer("cluster-count"))
+      this.map.addLayer({
+        id: "cluster-count",
+        type: "symbol",
+        source: "business",
+        filter: ["has", "point_count"],
+        layout: {
+          "text-field": "{point_count_abbreviated}",
+          "text-font": ["DIN Offc Pro Medium", "Arial Unicode MS Bold"],
```

Comment:
It's worth adding a comment here for posterity that Mapbox uses its own fonts and doesn't support this project's default Roboto.